src/table/components/head/ColumnAdjuster.tsx

```diff
@@ -13,17 +13,24 @@ import { focusHeadMenuButton } from '../../utils/accessibility-utils';
  * When you start dragging, mouse move and mouse up listeners are added.
  * While dragging the current column is updated, and on mouse up all other columns are updated.
  */
-const ColumnAdjuster = ({ column, isLastColumn, onColumnResize }: AdjusterProps) => {
+const ColumnAdjuster = ({ column, isLastColumn, onColumnResize, paginationTableRef }: AdjusterProps) => {
   const { pageColIdx } = column;
   const { rootElement, applyColumnWidths, constraints } = useContextSelector(TableContext, (value) => value.baseProps);
   const columnWidths = useContextSelector(TableContext, (value) => value.columnWidths);
   const setColumnWidths = useContextSelector(TableContext, (value) => value.setColumnWidths);
   const tempWidths = useRef({ adjusterHitArea: {}, columnWidth: 0, initX: 0, initWidth: 0 });
   const borderHeight = rootElement.getBoundingClientRect().height - PAGINATION_HEIGHT - BORDER_WIDTH;
 
+  const tableWidth = useContextSelector(TableContext, (value) => value.tableWidth);
+  const setBorderRight = useContextSelector(TableContext, (value) => value.setBorderRight);
+
   if (!applyColumnWidths || constraints.active) return null;
 
   const updateWidth = (adjustedWidth: number) => {
+    const columnsWidth = paginationTableRef?.current?.getBoundingClientRect?.()?.width;
+    if (columnsWidth && columnsWidth < tableWidth) setBorderRight(true);
+    else setBorderRight(false);
```

Comment:
I'm not sure we have to set this here. I think it is fine to set the borde once you are done changing the column width. That is when we update the other columns as well

Reply:
This also does not take into account if the table resizes and we should show/hide the border

Reply:
set the borde once you are done changing the column width. Yes, that's better.

Reply:
right, table resizes. Maybe `tableWidth={props.rect.width} `can help

Reply:
or you just calculate it on the table or context level. There is no point really setting it in the ColumnAdjuster. I'm pretty sure the value needs to change when columnWidths or rect.width changes.